analyzers/src/SonarAnalyzer.CSharp/Rules/ObsoleteAttributes.cs

```diff
@@ -21,7 +21,15 @@
 namespace SonarAnalyzer.Rules.CSharp;
 
 [DiagnosticAnalyzer(LanguageNames.CSharp)]
-public sealed class RemoveObsoleteCode : RemoveObsoleteCodeBase<SyntaxKind>
+public sealed class ObsoleteAttributes : ObsoleteAttributesBase<SyntaxKind>
```

Comment:
No need to make it plural. Make the class names singular (incl. filenames).
```suggestion
public sealed class ObsoleteAttribute : ObsoleteAttributeBase<SyntaxKind>
```

Reply:
Singular führt hier zu Namenskonflikten (das Attribut um das es geht heißt ja auch "ObsoleteAttribute"). Das würde dann zu ein paar voll qualifizierten Referenzen führen. Trotzdem konsquent für alle Dateien umsetzen?

Reply:
I didn't think of the name clash. Keep the name as is in this case.